src/avif.c

```diff
@@ -889,6 +889,13 @@ void avifCodecSpecificOptionsDestroy(avifCodecSpecificOptions * csOptions)
         avifFree(entry->key);
         avifFree(entry->value);
     }
+
+    csOptions->count = 0;
```

Comment:
Move the null pointer protection (lines 883-885) to the `avifCodecSpecificOptionsDestroy()` function.

Otherwise, passing NULL to `avifCodecSpecificOptionsDestroy()` will result in null pointer dereferencing in the `avifArrayDestroy()` call.

`avifCodecSpecificOptionsClear()` can assume its input is not NULL.